django/db/models/fields/files.py

```diff
@@ -295,20 +297,26 @@ def contribute_to_class(self, cls, name, **kwargs):
         setattr(cls, self.name, self.descriptor_class(self))
 
     def get_directory_name(self):
```

Comment:
A concern is that if a custom field overrides these deprecated methods, they'll now be ignored.